packages/core/src/lib/MockLog.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import {Log} from '..';
+
+/**
+ * An utility class to print nice Log messages.
+ */
+export class MockLog implements Log {
+    /**
+     * Creates a new Log instance
+     * @param tag the tag used when logging. Printed at the beggining of a log message.
+     * @param verbose if the Log is verbose. Debug messages are only printed on verbose logs.
+     */
+    public verbose: boolean;
+    private tag: string;
+    private receivedData: Array<string>;
+
+    constructor(tag = '', verbose = false) {
+      this.tag = tag;
+      this.verbose = verbose;
+      this.receivedData = [];
+    }
+
+    getReceivedData(): Array<string> {
+      return this.receivedData;
+    }
+
+    /**
+     * Prints a debug message to the Log. message is ignored if the Log is not set to verbose.
+     * @param message the message the be printed.
+     * @param args extra arguments for the console.
+     */
```

Comment:
@andreban do we need full docs for methods that override stuff for an interface?

Reply:
I'm happy with just keeping those on the interface.

Reply:
So should I remove all of them from ConsoleLog and MockLog?